sklearn/model_selection/_validation.py

```diff
@@ -765,11 +773,20 @@ def learning_curve(estimator, X, y, groups=None,
             clone(estimator), X, y, classes, train, test, train_sizes_abs,
             scorer, verbose) for train, test in cv.split(X, y, groups))
     else:
+        if shuffle:
+            rng = check_random_state(random_state)
+
+        train_test_proportions = []
+        for train, test in cv_iter:
+            for n_train_samples in train_sizes_abs:
+                if shuffle:
+                    train = rng.permutation(train)
+                train_test_proportions.append((train[:n_train_samples], test))
+
```

Comment:
I could have implemented it the following way:
```

        if shuffle:
            rng = check_random_state(random_state)
            train_test_proportions = [rng.permutation(train)[:n_train_samples], test)
                for train, test in cv for n_train_samples in train_sizes_abs]
        else:
            train_test_proportions = [(train[:n_train_samples], test)
                for train, test in cv for n_train_samples in train_sizes_abs]

```
but in this case line: `for train, test in cv for n_train_samples in train_sizes_abs]` is duplicated.